tests/rptest/tests/group_membership_test.py

```diff
@@ -113,3 +118,163 @@ def _check_no_duplicates():
+        gd = rpk.describe_group(group)
+        res = {}
+        for tp in gd.topics_partitions:
+            res[(tp.topic, tp.partition)] = tp.current_offset
+        return res
+
+    def _get_offset_from_metrics(self, group):
+        metric = self.redpanda.metrics_sample("kafka_group_offset")
+        if metric is None:
+            return None
+        metric = metric.label_filter(dict(group=group))
+        res = {}
+        for sample in metric.samples:
+            res[(sample.labels['topic'], sample.labels['partition'])] = sample.value
+        return res
+
+    def _wait_group_started_consuming(self, consumer, group):
+        rpk = RpkTool(self.redpanda)
+
+        def group_started_consuming():
+            return len(consumer.messages) > 0 and \
+                len(rpk.describe_group(group).topics_partitions) > 0 and \
+                rpk.describe_group(group).topics_partitions[0].current_offset > 0
+        wait_until(group_started_consuming, timeout_sec=20, backoff_sec=0.5)
+
+    @cluster(num_nodes=5)
+    def test_groups_offset_metric_check_value(self):
+        self.start_redpanda(num_nodes=3)
+        rpk = RpkTool(self.redpanda)
+
+        topic = "test_topic"
+        spec = TopicSpec(name=topic,
+                         partition_count=3)
+        self.redpanda.create_topic(spec)
+
+        # Wait for the partition to have a leader (`rpk produce` errors
+        # out if it tries to write data before this)
+        def partition_ready():
+            return KafkaCat(self.redpanda).get_partition_leader(
+                topic, 0)[0] is not None
+
+        wait_until(partition_ready, timeout_sec=10, backoff_sec=0.5)
+
+        num_records = 100
+        producer = KafProducer(self._ctx, self.redpanda, topic,
+                               num_records)
+        producer.start()
+
+        group = "g0"
+        consumer = RpkConsumer(self._ctx,
+                               self.redpanda,
+                               topic,
+                               group=group)
+        consumer.start()
+
+        self._wait_group_started_consuming(consumer, group)
+
+        def assert_second_offsets_larger(group_offsets_1, group_offsets_2):
+            assert len(group_offsets_1) == len(group_offsets_2)
+
+            for k, v in group_offsets_1.items():
+                assert k in group_offsets_2
+                assert v <= group_offsets_2[k]
+
+        rpk_offsets = self._get_offset_from_rpk_describe(group)
+
+        for i in range(10):
+            metrics_offsets = self._get_offset_from_metrics(group)
+            assert_second_offsets_larger(rpk_offsets, metrics_offsets)
+            rpk_offsets = self._get_offset_from_rpk_describe(group)
+            assert_second_offsets_larger(metrics_offsets, rpk_offsets)
+            time.sleep(10)
+
+        producer.stop()
+        consumer.stop()
+
+        rpk_offsets = self._get_offset_from_rpk_describe(group)
+        metrics_offsets = self._get_offset_from_metrics(group)
```

Comment:
There is a kafka cli tool (`kafka-consumer-groups.sh`) (and I think some of the functionality is available in`rpk` as well) which let's you precisely control the contents of a consumer group offset. this might be a way to create more precise and reliable tests rather than relying on consumers indirectly manipulating the consumer group state.

Reply:
Used  simple `rpk.produce` and `rpk.consume` to control offset